### PR TITLE
Remove approveAndCall

### PR DESCRIPTION
**Pull request by troggy**
So we can deposit ERC721 tokens

### CODE REVIEW

File: src/routes/deposit.js [hunk 1]
```diff
@@ -49,12 +49,10 @@ export default class Deposit extends React.Component {
       10 ** this.selectedToken.decimals
     );
 
-    bridge
-      .deposit(this.selectedToken, value)
-      .on('transactionHash', depositTxHash => {
-        console.log('deposit', depositTxHash); // eslint-disable-line
-        this.setState({ value: 0 });
-      });
+    bridge.deposit(this.selectedToken, value).then(({ transactionHash }) => {
```
Comment (sunify) [inline src/routes/deposit.js]: Promise will resolve only when the transaction will be mined. So field will be cleared after several seconds, but we need to clear it after transaction sending (when we have `transactionHash`)

Comment (troggy) [reply src/routes/deposit.js]: Fixing!

File: src/stores/token.ts [hunk 2]
```diff
@@ -117,4 +123,23 @@ export default class Token extends ContractStore {
...
+  * @param spender Account to approve transfer for
+  * @param value Minimal amount of allowance a spender should have
+  * @returns Promise resolved when allowance is enough for the transfer
+  */
+  private maybeApprove(spender: string, value: BigNumber) {
+    return this.iContract.methods.allowance(this.account.address, spender).call().then(allowance => {
+      if (Number(allowance) >= value.toNumber()) return;
+
+      const tx = this.iContract.methods
+        .approve(spender, new BigNumber(2 ** 255))
```
Comment (sunify) [inline src/stores/token.ts]: Why not `value − allowance`?

Comment (troggy) [reply src/stores/token.ts]: so that we have to execute `approve` only one time. Next deposit/bet will require just on tx

Comment (sunify) [reply src/stores/token.ts]: Yeah, but it looks like too invasive, no? It's ok for rinkeby, but not for mainnet

**Comment by sunify** [comment]
Looks nice 👍

File: src/stores/token.ts (hunk 2 above)
Comment (troggy) [reply src/stores/token.ts]: I thought so too, but some projects already doing that. For example, I was trading on https://radarrelay.com/ and they set the same DAI and WETH allowances for me. Not good (you have to keep trust), but better UX

Comment (sunify) [reply src/stores/token.ts]: Yeah, let's keep it for now.

I have idea for UI/UX, but i need to sleep with it :-)

Comment (troggy) [reply src/stores/token.ts]: > I have idea for UI/UX, but i need to sleep with it :-)

me too :)

**Pull request merged**